xc7/archs/artix7/CMakeLists.txt

```diff
@@ -0,0 +1,22 @@
+set(ARCH artix7)
+set(ROI_PART xc7a35tcpg236-1)
+set(ROI_DIR ${PRJXRAY_DB_DIR}/${ARCH}/harness/basys3/swbut)
+
+project_xray_prepare_database(
+  PART ${ARCH}
+)
+
+add_subdirectory(tiles)
+add_subdirectory(blocks)
+
+add_xc7_arch_define(
```

Comment:
How does this function not have an ARCH ${ARCH} argument?
Should project_xray_prepare_database just move within add_xc7_arch_define?

Reply:
The ${ARCH} variable is set in the specific archs folder (e.g. artix7 or zynq7) being them in different scopes the ${ARCH} variable should be consistent with each different architecture. I found it easier this way because that is a variable that is used consistently within all the various subdirectories

Reply:
That's exactly what you shouldn't do.  Attach the variable to a target name if you must, but do not rely on scoped variables like that.  It is hard to debug and prevents refactoring.

Reply:
Ok, good to now, I have re-added the ARCH argument.